overlord/hookstate/ctlcmd/set.go

```diff
@@ -42,6 +52,21 @@ func (s *setCommand) Execute(args []string) error {
 		return fmt.Errorf("cannot set without a context")
 	}
 
-	// TODO: Talk to the handler to take care of the set request.
+	for _, patchValue := range s.Positional.ConfValues {
+		parts := strings.SplitN(patchValue, "=", 2)
+		if len(parts) != 2 {
+			return fmt.Errorf(i18n.G("invalid configuration: %q (want key=value)"), patchValue)
+		}
+		key := parts[0]
+		var value interface{}
+		err := json.Unmarshal([]byte(parts[1]), &value)
+		if err != nil {
+			// Not valid JSON-- just save the string as-is.
+			value = parts[1]
+		}
+
+		s.context().Handler().SetConf(key, value)
```

Comment:
Done. This is now setting it in a transaction directly, and storing it in the `Context` cache. It utilizes the `Context`'s new-found deferring capability to `Commit()` the transaction after it's completed successfully.